docs/pages/guides/react-compiler.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: React Compiler
+description: Learn how to enable and use the React Compiler in Expo apps.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+import { Step } from '~/ui/components/Step';
+
+> Warning: React Compiler is still experimental!
+
+The new [React Compiler](https://react.dev/learn/react-compiler) automatically memoizes components and hooks to enable fine-grained reactivity. This can lead to significant performance improvements in your app. The React Compiler is currently experimental and is not enabled by default. You can enable it in your app by following the instructions below.
+
+## Enabling React Compiler
+
+First, [check how compatible](https://react.dev/learn/react-compiler#checking-compatibility) your project is with the React Compiler.
+
+<Terminal cmd={['$ npx react-compiler-healthcheck@latest']} />
+
+This will generally verify if your app is following the [**rules of React**](https://react.dev/reference/rules).
+
+Next, toggle on the React Compiler experiment in your `app.json` file:
+
+```json app.json
+{
+  "experiments": {
+    "reactCompiler": true
+  }
+}
+```
+
+### Enabling the linter
+
+Additionally, you should use the ESLint plugin to continuously enforce the rules of React in your project.
+
+Do this by running `npx expo lint` to ensure Lint is setup in your app, then install the ESLint plugin:
+
+<Terminal cmd={['$ npx expo install eslint-plugin-react-compiler']} />
+
+Then update your ESLint configuration to include the plugin:
+
+```js .eslintrc.js
+// https://docs.expo.dev/guides/using-eslint/
+module.exports = {
+  extends: 'expo',
+  plugins: ['eslint-plugin-react-compiler'],
+  rules: {
+    'react-compiler/react-compiler': 'error',
+  },
+};
+```
+
+> In the future, all of these steps will be automated by Expo CLI.
+
+## Incremental adoption
+
+You can incrementally adopt the React Compiler in your app using a few strategies:
+
+1. Configure the Babel plugin to only run on specific files or components.
+
+To do this, add the following to your Babel configuration:
+
+```js babel.config.js
+module.exports = function (api) {
+  api.cache(true);
+
+  return {
+    presets: [
+      [
+        'babel-preset-expo',
+        {
+          'react-compiler': {
+            sources: filename => {
+              // Match file names to include in the React Compiler.
+              return filename.includes('src/path/to/dir');
+            },
+          },
+        },
+      ],
+    ],
+  };
+};
+```
+
+Whenever you change your `babel.config.js` file, you need to restart the Metro bundler to apply the changes:
+
+<Terminal cmd={['$ npx expo start --clear']} />
+
+2. Use the `"use no memo"` directive to opt-out of the React Compiler for specific components or files.
+
+```jsx
+function MyComponent() {
+  'use no memo';
+
+  return <Text>Will not be optimized</Text>;
+}
+```
```

Comment:
````suggestion
<Step label="1">

Configure the Babel plugin to only run on specific files or components. To do this, add the following to your project's Babel configuration:

```js babel.config.js
module.exports = function (api) {
  api.cache(true);

  return {
    presets: [
      [
        'babel-preset-expo',
        {
          'react-compiler': {
            sources: filename => {
              // Match file names to include in the React Compiler.
              return filename.includes('src/path/to/dir');
            },
          },
        },
      ],
    ],
  };
};
```

Whenever you change your **babel.config.js** file, you need to restart the Metro bundler to apply the changes:

<Terminal cmd={['$ npx expo start --clear']} />

</Step>

<Step label="2">

Use the `"use no memo"` directive to opt out of the React Compiler for specific components or files.

```jsx
function MyComponent() {
  'use no memo';

  return <Text>Will not be optimized</Text>;
}
```

</Step>
````

- Let's use `Step` to describe procedure.
- Update Filenames to be a bold text